### PR TITLE
refactor: Forms 1119 profile settings composition api

### PR DESCRIPTION
**Pull request by jasonchung1871**
# Description
The Form Settings have been updated from the options API to the composition API and have 100% test coverage on them.

The transformUtils function getAvailableDates that is supposed to grab available dates for a submission period would not always return the same value, as created tests showed. I've rewritten the functions with better documentation and a refactored input. It should still output the same values but it is unknown as the previous documentation for this feature does not actually specify the inputs/outputs nor were there any tests.

The outputs should be the same as before but I'm unsure since there wasn't any previous documentation on inputs/outputs nor tests. It could be breaking but as far as I know, it is working.

Resolves: https://bcdevex.atlassian.net/jira/software/c/projects/FORMS/boards/95?selectedIssue=FORMS-1119

## Types of changes

Bug fix (non-breaking change which fixes an issue)

Documentation (non-breaking change with enhancements to documentation)
Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

### CODE REVIEW

File: app/frontend/src/components/designer/profile/FormLabelProfile.vue [hunk 1]
```diff
@@ -13,12 +13,14 @@ const form = computed(() => formStore.form);
 const lang = computed(() => formStore.lang);
 const isRTL = computed(() => formStore.isRTL);
 
+/* c8 ignore start */
```
Comment (jasonchung1871) [inline app/frontend/src/components/designer/profile/FormLabelProfile.vue:16]: Maybe we could add coverage for the front end validation rules..

File: app/frontend/src/utils/transformUtils.js [hunk 2]
```diff
@@ -75,167 +75,96 @@ export function attachAttributesToLinks(formSchemaComponents) {
...
+ * @returns {Object} An object array of available dates in given period
  */
-export function getAvailableDates(
-  keepAliveFor = 0,
-  keepAliveForInterval = 'days',
-  formStartDate,
-  term = null,
-  interval = null,
+export function getSubmissionPeriodDates(
+  keepOpenForTerm = 0,
```
Comment (jasonchung1871) [inline app/frontend/src/utils/transformUtils.js:92]: Renamed these variables to match what is actually found in the form.schedule object

File: app/frontend/src/utils/transformUtils.js [hunk 3]
```diff
@@ -75,167 +75,96 @@ export function attachAttributesToLinks(formSchemaComponents) {
...
- * @param {Integer} keepOpenForTerm keepOpenForTerm An integer of number of Days/Weeks OR Years, that tells form to be open for a particular period
- * @param {String} keepOpenForInterval keepOpenForInterval A string of days,Weeks,months, that tells form to be open for a particular period
- * @param {Integer} allowLateTerm An integer of number of Days/Weeks OR Years, that tells form to be allowed for late submission for a particular period
- * @param {String} allowLateInterval A string of days,Weeks,months, that tells form to be allowed for late submission for a particular period
- * @param {Integer} repeatSubmissionTerm An integer of number of Days/Weeks OR Years, that tells form to be repeated for a particular period
- * @param {String} repeatSubmissionInterval A string of days,Weeks,months, that tells form to be repeated for a particular period
- * @param {Object[]} repeatSubmissionUntil An object of Moment JS date, that tells form to be finally close after repetition on particular day date
- * @returns {Object[]} An object of Moment JS date
- */
-export function getCalculatedCloseSubmissionDate(
```
Comment (jasonchung1871) [inline app/frontend/src/utils/transformUtils.js:179]: This whole function is currently unnecessary.

**Review by vijaivir — APPROVED** [review]
Nice work!

**Pull request merged**